### DISCUSSION TITLE
Unable to create identity on duplicated field.

### DISCUSSION THREAD

**Discussion by ushenkodmitry** [JasperFx/marten#1776] [Q&A]
Hello,
I don't know if it's bug or not. And I can't find any information on this.
When I use duplicate field as identity Marten doesn't generate a column for that field in table creation DDL. Is this a bug, or this is by design? 
The code I use is simple 
```
For<User>()
.Duplicate(e => e.Id)
.Identity(e => e.Id);
```
Marten doesn't generate column in table but generates scrip for index creation on that field. As result I'm catching PostgresException.

**Comment by jeremydmiller** [comment]
@ushenkodmitry That's redundant anyway. The identity is always separated into an "id" field in the document tables, and becomes your primary key. Also redundant because Marten automatically makes the "Id" property or field the identity member.

**Reply by ushenkodmitry** [reply]
@jeremydmiller 
With Id property in place I don't see any exceptions. Got it.
But what if want to use another property as identity column? Can I configure it with Identity method? In the docs I see statements that are saying that I can:

> If you really want to -- or you're migrating existing document types from another document database -- Marten provides the [Identity] attribute to force Marten to use a property or field as the identifier that doesn't match the "id" or "Id" or "ID" convention

**Comment by ushenkodmitry** [comment]
Wihout setting identity column explicitly with Identity method this code
throws exception. Don't remember what type exactly, but the message informs
me that the type does't have identity column configured.

вс, 21 мар. 2021 г., 17:54 Jeremy D. Miller ***@***.***>:

**Reply by jeremydmiller** [reply]
And that's described in the documentation: 
https://martendb.io/documentation/documents/identity/
